### PR TITLE
AM and MessageHelper unit tests adapted after porting

### PR DESCRIPTION
**Pull request by dev-gh**
Disabled tests:
- ApplicationImplTest LoadPersistentFiles
- HMICapabilitiesTest LoadCapabilitiesFromFile

### CODE REVIEW

**Comment by dev-gh** [comment]
@OHerasym @anosach-luxoft 
Please review

**Comment by OHerasym** [comment]
reviewed

File: src/components/application_manager/src/resumption/resumption_sql_queries.cc [hunk 1]
```diff
@@ -428,7 +428,7 @@ const std::string kUpdateHMILevel =
 
 const std::string kUpdateIgnOffCount =
     "UPDATE `application` "
-    "SET `ign_off_count` = `ign_off_count` - 1"
+    "SET `ign_off_count` = `ign_off_count` - 1 "
```
Comment (anosach-luxoft) [inline src/components/application_manager/src/resumption/resumption_sql_queries.cc:431]: @dev-gh please revert this change

File: src/components/application_manager/test/hmi_capabilities_test.cc [hunk 2]
```diff
@@ -342,29 +342,6 @@ TEST_F(HMICapabilitiesTest, LoadCapabilitiesFromFile) {
   EXPECT_EQ("SE", vehicle_type_so["trim"].asString());
 }
 
-TEST_F(HMICapabilitiesTest, ConvertJsonLanguagesToObj) {
```
Comment (anosach-luxoft) [inline src/components/application_manager/test/hmi_capabilities_test.cc:345]: @dev-gh mb just disable it?

Comment (dev-gh) [inline src/components/application_manager/test/hmi_capabilities_test.cc:345]: @anosach-luxoft 
This interface has been removed so I don't think that we have to test it than.

File: src/components/application_manager/test/resumption_sql_queries_test.cc [hunk 3]
```diff
@@ -84,7 +84,9 @@ const string kDeleteData =
     "COMMIT; "
     "VACUUM;";
 
-const std::string kJournalOff = "PRAGMA journal_mode = OFF;";
+const static std::string kJournalOff = "PRAGMA journal_mode = OFF;";
```
Comment (anosach-luxoft) [inline src/components/application_manager/test/resumption_sql_queries_test.cc:86]: @dev-gh what for `static` specificator?

**Comment by anosach-luxoft** [comment]
@VVeremjova, @AByzhynar, @Kozoriz please review.
This PR contains changes and review notes fixes from [previous PR](https://github.com/anosach-luxoft/sdl_core/pull/17).

File: src/components/application_manager/test/hmi_capabilities_test.cc (hunk 2 above)
Comment (anosach-luxoft) [inline src/components/application_manager/test/hmi_capabilities_test.cc:345]: @dev-gh ok

**Comment by anosach-luxoft** [comment]
@dev-gh BTW you have to enable AM and MH tests build in `application_manager/CMakeLists.txt` file

File: src/components/application_manager/src/resumption/resumption_sql_queries.cc (hunk 1 above)
Comment (dev-gh) [inline src/components/application_manager/src/resumption/resumption_sql_queries.cc:431]: @anosach-luxoft 
I believe this is an issue, I think query will be incorrect than.

File: src/components/application_manager/test/resumption_sql_queries_test.cc (hunk 3 above)
Comment (dev-gh) [inline src/components/application_manager/test/resumption_sql_queries_test.cc:86]: @anosach-luxoft 
As I can see its being used in SetUpTestCase().

File: src/components/application_manager/src/resumption/resumption_sql_queries.cc (hunk 1 above)
Comment (anosach-luxoft) [inline src/components/application_manager/src/resumption/resumption_sql_queries.cc:431]: @dev-gh no. I see `"SET 'ign_off_count' = 'ign_off_count' + 1"` (without space at the end) some below

Comment (anosach-luxoft) [inline src/components/application_manager/src/resumption/resumption_sql_queries.cc:431]: @dev-gh my mistake. Space is needed

**Comment by dev-gh** [comment]
@anosach-luxoft 
I'm on different commit - test are enabled already.

File: src/components/application_manager/src/resumption/resumption_sql_queries.cc (hunk 1 above)
Comment (OHerasym) [inline src/components/application_manager/src/resumption/resumption_sql_queries.cc:431]: @anosach-luxoft , @dev-gh , without space, Exec returns false, and one test fails.

**Comment by VVeremjova** [comment]
@dev-gh Reviewed

File: src/components/application_manager/src/resumption/resumption_sql_queries.cc (hunk 1 above)
Comment (AByzhynar) [inline src/components/application_manager/src/resumption/resumption_sql_queries.cc]: +1 to @OHerasym

File: src/components/application_manager/test/resumption_sql_queries_test.cc (hunk 3 above)
Comment (AByzhynar) [inline src/components/application_manager/test/resumption_sql_queries_test.cc:86]: @dev-gh May be just use nameless namespace without adding `static` qualifier? I mean to wrap these strings.

File: src/components/utils/src/threads/thread_qt.cc [hunk 4]
```diff
@@ -61,6 +61,9 @@ void* Thread::threadFunc(void* arg) {
   threads::Thread* thread = static_cast<Thread*>(arg);
   DCHECK(thread);
 
+  // Sets thread id in order to be able to check that id on thread joining
+  thread->set_thread_handle(QThread::currentThread);
```
Comment (pestOO) [inline src/components/utils/src/threads/thread_qt.cc]: Looks like you are setting method as a handle.

File: src/components/include/utils/threads/thread.h [hunk 5]
```diff
@@ -216,6 +216,16 @@ class Thread {
     return handle_;
   }
 
+#if defined(QT_PORT)
+  /**
+   * @brief Sets handle of thread which has delegate running
+   * @param handle Thread handle
+   */
+  void set_thread_handle(PlatformThreadHandle handle) {
```
Comment (pestOO) [inline src/components/include/utils/threads/thread.h]: Seems this method need to be hide as a private because of internal usage only.

File: src/components/utils/src/threads/thread_qt.cc (hunk 4 above)
Comment (anosach-luxoft) [inline src/components/utils/src/threads/thread_qt.cc]: @dev-gh you have to protect `handle_` variable, so move it down after lock Acquire

File: src/components/utils/src/threads/thread_qt.cc [hunk 6]
```diff
@@ -113,11 +116,14 @@ bool Thread::start() {
...
-  return QThread::currentThread();
+  // Available interfaces QThread::currentThread() and
+  // QThread::currentThreadId() return values, which can't be converted to
+  // uint64_t, thus using native WIN interface.
+  return static_cast<uint64_t>(GetCurrentThreadId());
 }
 
 bool Thread::IsCurrentThread() const {
-  return CurrentId() == thread_handle();
+  return QThread::currentThread() == thread_handle();
```
Comment (anosach-luxoft) [inline src/components/utils/src/threads/thread_qt.cc:128]: @dev-gh now you're going to compare pointers, mb would be better to compare `uint64_t` values?

File: src/components/include/utils/threads/thread.h [hunk 7]
```diff
@@ -216,6 +216,16 @@ class Thread {
     return handle_;
   }
 
+#if defined(QT_PORT)
```
Comment (anosach-luxoft) [inline src/components/include/utils/threads/thread.h]: @dev-gh I think this setter is redundant. You're able to set `handle_` inside `threadFunc` even without it

File: src/components/utils/src/threads/thread_qt.cc (hunk 6 above)
Comment (dev-gh) [inline src/components/utils/src/threads/thread_qt.cc:128]: @anosach-luxoft 
I see that conversion to uint64_t is necessary only once - in CurrentId(), so I would prefer to use native QT handle instead of converted value.

File: src/components/utils/src/threads/thread_qt.cc (hunk 4 above)
Comment (dev-gh) [inline src/components/utils/src/threads/thread_qt.cc]: @anosach-luxoft @pestOO 
Done with   1af44d5

File: src/components/include/utils/threads/thread.h (hunk 7 above)
Comment (dev-gh) [inline src/components/include/utils/threads/thread.h]: @anosach-luxoft 
Done with   1af44d5

File: src/components/include/utils/threads/thread.h (hunk 5 above)
Comment (dev-gh) [inline src/components/include/utils/threads/thread.h]: @pestOO 
Removed as unnecessary with     1af44d5
